app/src/chat_engine.py

```diff
@@ -52,14 +53,16 @@ def create_engine(engine_id: str) -> ChatEngineInterface | None:
 class GuruBaseEngine(ChatEngineInterface):
     def on_message(self, question: str) -> OnMessageResult:
         with db.PostgresDBClient().get_session() as db_session:
-            chunks = retrieve(
+            chunks = retrieve_with_scores(
```

Comment:
Crux of my feedback: `retrieve_with_scores()` returns `Sequence[Row[Tuple[Chunk, float]]]`, whereas `retrieve()` returns a basic `Sequence[Chunk]`. This means `chunks` is now a misnomer -- it should be `chunks_with_scores`. To facilitate future development, anywhere you use `chunk[1]` should be renamed `score`, and `chunk[0]` is the original `chunk`.